images/hub/jupyterhub_config.py

```diff
@@ -177,6 +178,25 @@ def get_config(key, default=None):
     c.DummyAuthenticator.password = get_config('auth.dummy.password', None)
 elif auth_type == 'tmp':
     c.JupyterHub.authenticator_class = 'tmpauthenticator.TmpAuthenticator'
+elif auth_type == 'custom':
+    # full_class_name looks like "myauthenticator.MyAuthenticator".
+    # To create a docker image with this class availabe, you can just have the
+    # following Dockerifle:
+    #   FROM jupyterhub/k8s-hub:v0.4
+    #   RUN pip3 install myauthenticator
+    full_class_name = get_config('auth.custom.class_name')
+    c.JupyterHub.authenticator_class = full_class_name
+
+    (module, class_name) = full_class_name.split('.')
+    the_module = importlib.import_module(module)
+    klass=getattr(the_module, class_name)
+
+    class_params = get_config('auth.custom.params')
```

Comment:
I looked at traitlets and I couldn't find anything that would update a class' traitlets in one step.

There's `update_config` which maybe does this, but it works on an instance:
https://github.com/ipython/traitlets/blob/master/traitlets/config/configurable.py#L206

I looked at the `Configurable` and `HasTraits` classes.

If you want this code to be more traitlets oriented, I can use `has_trait` and `set_trait` instead of Python's `hasattr` and `setattr`:
https://github.com/ipython/traitlets/blob/master/traitlets/traitlets.py#L1432
https://github.com/ipython/traitlets/blob/master/traitlets/traitlets.py#L1369